### PR TITLE
	revert revert of test/e2e/generated/BUILD

### PR DESCRIPTION
**Pull request by mikedanese**
broken in fad1990

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37114)

### CODE REVIEW

**Comment by mikedanese** [comment]
This fixes bazel builds in the release branch.

**Comment by ixdy** [comment]
/lgtm

@k8s-bot bazel build this

**Comment by mikedanese** [comment]
@k8s-bot bazel build this

File: test/e2e/BUILD [hunk 1]
```diff
@@ -210,20 +210,13 @@ go_library(
     ],
 )
 
-go_test(
+# phony go_default_test. Bazel test run interprets these as
```
Comment (ixdy) [inline test/e2e/BUILD]: should these be go_binarys instead? our e2e infrastructure expects an `e2e_test` binary, I think.

Comment (mikedanese) [inline test/e2e/BUILD]: Hmm, good idea. I will see if that works

Comment (mikedanese) [inline test/e2e/BUILD]: I can't fix this without modifying rules_go. I will add it to the list of stuff to finish.

**Comment by mikedanese** [comment]
@k8s-bot bazel build this

File: test/e2e/BUILD [hunk 2]
```diff
@@ -210,20 +210,23 @@ go_library(
     ],
 )
 
-go_test(
+# this is a private library until we can support building the
+# test binary
+# TODO(mikedanese): fix @io_bazelbuild/rules_go
```
Comment (ixdy) [inline test/e2e/BUILD]: do you have an issue tracking this?

File: test/e2e/BUILD [hunk 3]
```diff
@@ -210,20 +210,23 @@ go_library(
...
     library = "go_default_library",
     tags = ["automanaged"],
+    visibility = ["//visibility:private"],
     deps = [
         "//pkg/api:go_default_library",
         "//pkg/client/clientset_generated/internalclientset:go_default_library",
         "//pkg/metrics:go_default_library",
         "//test/e2e/framework:go_default_library",
         "//vendor:github.com/onsi/ginkgo",
-        "//vendor:github.com/onsi/gomega",
```
Comment (ixdy) [inline test/e2e/BUILD]: why remove this? `metrics_grabber_test.go` still seems to directly depend on it.

Comment (mikedanese) [inline test/e2e/BUILD]: I was testing if gazel would still correct this. It does. Fixed.

File: test/e2e/BUILD (hunk 2 above)
Comment (mikedanese) [inline test/e2e/BUILD]: https://github.com/kubernetes/kubernetes/issues/37121

**Comment by mikedanese** [comment]
@k8s-bot bazel build this

**Comment by mikedanese** [comment]
That was from removing the import.

**Comment by mikedanese** [comment]
@k8s-bot bazel build this

**Review by ixdy — APPROVED** [review]
/lgtm

**Pull request merged**